lib/engine/game/g_18_zoo.rb

```diff
@@ -0,0 +1,668 @@
+      def location_name(coord)
+        self.class::LOCATION_NAMES_BY_MAP[@map][coord]
+      end
+
+      def self.title
+        '18ZOO'
+      end
+
+      def unowned_purchasable_companies(_entity)
+        @available_companies + @future_companies
+      end
+
+      private
+
+      def init_round
+        Round::Draft.new(self, [Step::G18ZOO::SimpleDraft], reverse_order: true)
+      end
+
+      def num_trains(train)
+        num_players = @players.size
+
+        case train[:name]
+        when '2S'
+          num_players
+        else
+          super
+        end
+      end
+
+      def init_corporations(stock_market)
+        self.class::CORPORATIONS.select { |corporation| CORPORATIONS_BY_MAP[@map].include?(corporation[:sym]) }
+                                .map do |corporation|
+          Corporation.new(
+            min_price: stock_market.par_prices.map(&:price).min,
+            capitalization: self.class::CAPITALIZATION,
+            coordinates: CORPORATION_COORDINATES_BY_MAP[@map][corporation[:sym]],
+            **corporation.merge(corporation_opts),
+          )
+        end
+      end
+
+      def init_starting_cash(players, bank)
+        hash = SMALL_MAP.include?(@map) ? self.class::STARTING_CASH_SMALL_MAP : self.class::STARTING_CASH_BIG_MAP
+        cash = hash[players.size]
+
+        players.each do |player|
+          bank.spend(cash, player)
+        end
+      end
+
+      def custom_end_game_reached?
+        @turn == 3
+      end
+
+      def new_stock_round
+        result = super
+
+        add_cousins if @turn == 3
+
+        update_current_and_future(@companies_for_monday, @companies_for_tuesday, 1)
+        update_current_and_future(@companies_for_tuesday, @companies_for_wednesday, 2)
+        update_current_and_future(@companies_for_wednesday, nil, 3)
+
+        @available_companies.select { |c| c.owner.nil? }.each { |c| c.owner = @bank }
+
+        result
+      end
+
+      def new_operating_round(round_num = 1)
+        @operating_rounds = 3 if @turn == 3 # Last round has 3 ORs
+
+        super
+      end
+
+      def operating_round(round_num)
+        Round::Operating.new(self, [
+          Step::Track,
+          Step::Token,
+        ], round_num: round_num)
+      end
+
+      def round_description(name, round_number = nil)
+        round_number ||= @round.round_num
+        day = case @turn
+              when 1
+                'Monday'
+              when 2
+                'Tuesday'
+              when 3
+                'Wednesday'
+              end
+
+        case name
+        when 'Draft'
+          name
+        when 'Stock'
+          "#{day} Stock"
+        when 'Operating'
+          "#{day} Operating Round (#{round_number} of #{@operating_rounds})"
+        end
+      end
+
+      # Override this, and add elements (paragraphs of text) here to display it on Info page.
+      def timeline
+        @timeline.append("Game ends after OR #{@last_or}")
```

Comment:
@timeline <<

Reply:
I removed it, the text was not correct anyway.